### PR TITLE
Preview: Add missing trait impls to make rustix/nix compile

### PR DESCRIPTION
**Pull request by flba-eb**
Dear @gh-tr, @samkearney and @Tom-Goring,

Before I make this PR public, feel free to have a look at these changes. They are required to make `rustix`, `nix`, `clap-4.2` and `ctrlc` compile.

### CODE REVIEW

File: src/unix/mod.rs [hunk 1]
```diff
@@ -1499,6 +1499,11 @@ cfg_if! {
                 timeout: *mut timespec,
                 sigmask: *const sigset_t,
             ) -> ::c_int;
+            pub fn sigaction(
```
Comment (gh-tr) [inline src/unix/mod.rs:1502]: Is it alright that the signature doesn't match for QNX?  I assume this came from another Linux implementation:
  extern int sigaction(int __signo, const struct sigaction *__act, struct sigaction *__oact);

File: src/unix/nto/mod.rs [hunk 2]
```diff
@@ -641,7 +668,9 @@ s_no_extra_traits! {
 
     pub struct sigevent {
```
Comment (gh-tr) [inline src/unix/nto/mod.rs:669]: Why was the extra padding added?  The structure itself is 32 bytes on 64 arches on QNX.  With these changes, the structure is larger than our internal implementation.  For a safety focused language, this seems like a strange thing to do.

**Review by gh-tr — COMMENTED** [review]
Couple of questions inline

File: src/unix/mod.rs (hunk 1 above)
Comment (flba-eb) [reply src/unix/mod.rs:1502]: @gh-tr  I fail to see the difference. The parameter names are different but the types should be all the same. As with C/C++, the parameter name should not be relevant (I don't consider the parameter **name** as part of the signature).

File: src/unix/nto/mod.rs (hunk 2 above)
Comment (flba-eb) [reply src/unix/nto/mod.rs:669]: The size should not have changed. The second member `__sigev_un1` with type usize (8 bytes with 8-byte-alignment) does not actually exist on `C` side but is a union. I have removed `__sigev_un1`.

The union contains an `int` (4 bytes) as signal number and other members of pointer type (64 bit aligned on 64 bit machines). If I'm not mistaken, all members of the union start at the same address. It is the address of the member with the highest alignment requirements, i.e. the signal number gets an 8-byte-alignment (even though 4 bytes would be sufficient for an int). The `padding1` member ensures this alignment.
The alignment and struct size is also confirmed by the ctest2 check.

What is your suggestion to make the signal number accessible? What did I miss?

Comment (flba-eb) [reply src/unix/nto/mod.rs:669]:
```text
    //         |4bytes     |4bytes     |4bytes     |4bytes   |4bytes   |
    // before: |notify     | padding(*)|    un1 (usize)      |value    |
    // after : |notify     | pad1      | signo     | pad2    |value    |
```
(*) Caused by alignment requirement of next member un1 (usize)

**Comment by flba-eb** [comment]
upstream pull request can be created

**Comment by flba-eb** [comment]
See https://github.com/rust-lang/libc/pull/3273

**Pull request closed without merging**